client/landing/stepper/hooks/use-site-migration-key.ts

```diff
@@ -10,12 +10,14 @@ const getMigrationKey = async ( siteId: number ): Promise< ApiResponse > =>
 		apiNamespace: 'wpcom/v2',
 	} );
 
-export const useSiteMigrationKey = ( siteId?: number ) => {
+type Options = Pick< UseQueryOptions, 'enabled' >;
+
+export const useSiteMigrationKey = ( siteId?: number, options?: Options ) => {
```

Comment:
Maybe I didn't explain myself correctly. Basically I don't see the value of picking from the `UseQueryOption` type since it is not going to affect the query directly, making the semantics slightly different.

Say you have the following code:
```useSiteMigrationKey( null, { enabled: true } )```

 I would expect for the actual API call to be made, because I enabled it and it's a UseQueryOption pick, so it would be reasonable to override default behaviour. But it is not made because siteId is null.
If we had a custom type for this hook, say something like:
```
interface Options {
   stalling: boolean?
}
``` 
This purpose of this call  `useSiteMigrationKey( null, { stalling: true } )` would be clearer to me, because I wouldn't have an expectation about the inner implementation.

Reply:
I gotcha, you won't to create false expectations by saying this ` enabled` will not work exactly in a similar way to the original one.

I have mixed feelings, so I would like to think more.

Do you mind if I create a separate issue just to give me more time to think about it and not block it?

Reply:
> I gotcha, you won't to create false expectations by saying this  enabled will not work exactly in a similar way to the original one.

In my mind, the expectation is created by the `enabled` being picked from `UseQueryOption` not from it being called `enabled`.

> Do you mind if I create a separate issue just to give me more time to think about it and not block it?
yup, go ahead! I'll approve all of this